src/java-interop/java-interop-gc-bridge-mono.c

```diff
@@ -262,6 +262,10 @@ java_interop_gc_bridge_new (JavaVM *jvm)
 	if (jvm == NULL)
 		return NULL;
 
+	if (!monodroid_dylib_mono_init (monodroid_get_dylib (), NULL)) {
```

Comment:
...how does this *build and run* on the PR builder?

That this works at all without `#if` is blowing my mind.

Looking at the [build log](https://jenkins.mono-project.com/job/Java.Interop-pr-builder/475/), we're seeing the warning I would expect to see:

```
java-interop-gc-bridge-mono.c(265,7): warning G84A3CD4A: implicit declaration of function 'monodroid_dylib_mono_init' is invalid in C99 [-Wimplicit-function-declaration] [/Users/builder/jenkins/workspace/Java.Interop-pr-builder/src/java-interop/java-interop.csproj]
          if (!monodroid_dylib_mono_init (monodroid_get_dylib (), NULL)) {
               ^
```

Good thing we don't have "warning as errors" enabled! ;-)

Reading further, my mind is no longer blown: it's blowing up during unit tests:

```
  mono --debug packages/NUnit.ConsoleRunner.3.7.0/tools/nunit3-console.exe  bin/TestDebug/Java.Interop-Tests.dll  --result="TestResult-Java.Interop-Tests.xml;format=nunit2" --output="bin/TestDebug/TestOutput-Java.Interop-Tests.txt"
...
  Test Files
      bin/TestDebug/Java.Interop-Tests.dll
  
  dyld: lazy symbol binding failed: Symbol not found: _monodroid_get_dylib
    Referenced from: /Users/builder/jenkins/workspace/Java.Interop-pr-builder/bin/TestDebug/libjava-interop.dylib
    Expected in: flat namespace
  
  dyld: Symbol not found: _monodroid_get_dylib
    Referenced from: /Users/builder/jenkins/workspace/Java.Interop-pr-builder/bin/TestDebug/libjava-interop.dylib
...
/Users/builder/jenkins/workspace/Java.Interop-pr-builder/build-tools/scripts/RunNUnitTests.targets(30,5): error MSB3073: The command "mono --debug packages/NUnit.ConsoleRunner.3.7.0/tools/nunit3-console.exe  bin/TestDebug/Java.Interop-Tests.dll  --result="TestResult-Java.Interop-Tests.xml;format=nunit2" --output="bin/TestDebug/TestOutput-Java.Interop-Tests.txt"" exited with code 156.
  Build continuing because "ContinueOnError" on the task "Exec" is set to "ErrorAndContinue".
Done Building Project "/Users/builder/jenkins/workspace/Java.Interop-pr-builder/build-tools/scripts/RunNUnitTests.targets" (default targets) -- FAILED.

Build FAILED.
```

Given that it clearly failed...***WHY DIDN'T IT FAIL?!***

Now my mind is blown, just differently from before. :-(

With that exploration out of the way...

Reply:
This `if` block needs to be wrapped in `#if defined (ANDROID) || defined (DYLIB_MONO)`:

```c
#if defined (ANDROID) || defined (DYLIB_MONO)
	if (!monodroid_dylib_mono_init (monodroid_get_dylib (), NULL)) {
		log_fatal (LOG_DEFAULT, "mono runtime initialization error: %s", dlerror ());
		exit (FATAL_EXIT_CANNOT_FIND_MONO);
	}
#endif  /* defined (ANDROID) || defined (DYLIB_MONO) */
```

Reply:
Doh, indeed. At least we have the PR builder fixed as well. Good catch!